ultrasonics/official_plugins/up_spotify.py

```diff
@@ -555,7 +555,11 @@ def saved_songs_add(self, spotify_ids):
             songs_dict = []
 
             for playlist in playlists:
-                item = {"name": playlist["name"], "id": {"spotify": playlist["id"]}}
+                if playlist is not None and isinstance(playlist, dict):
+                    name = playlist.get("name")
+                    playlist_id = playlist.get("id")
+                    if name is not None and playlist_id is not None:
+                        item = {"name": name, "id": {"spotify": playlist_id}}
 
```

Comment:
`item` should only be added if all the if checks above pass - should be nested within the last check